src/router/compute_test.go

```diff
@@ -355,3 +356,101 @@ func TestRouterComputeSingleError(t *testing.T) {
 		assert.NotNil(t, err)
 	}
 }
+
+func TestRouterComputeListError(t *testing.T) {
+	log := xlog.NewStdLog(xlog.Level(xlog.PANIC))
+	router, cleanup := MockNewRouter(log)
+	defer cleanup()
+
+	// Shardkey is NULL.
+	{
+		assert.NotNil(t, router)
+		_, err := router.ListUniform("t1", "", sqlparser.PartitionOptions{})
+		assert.NotNil(t, err)
+	}
+
+	// Table is NULL.
+	{
+		assert.NotNil(t, router)
+		_, err := router.ListUniform("", "i", sqlparser.PartitionOptions{})
+		assert.NotNil(t, err)
+	}
+
+	// different backends with the same list value.
+	{
+		partitionDef := sqlparser.PartitionOptions{
+			&sqlparser.PartitionDefinition{
+				Backend: "node1",
+				Row:     sqlparser.ValTuple{sqlparser.NewStrVal([]byte("1"))},
+			},
+			&sqlparser.PartitionDefinition{
+				Backend: "node2",
+				Row:     sqlparser.ValTuple{sqlparser.NewIntVal([]byte("1"))},
+			},
+		}
+
+		assert.NotNil(t, router)
+		_, err := router.ListUniform("t1", "i", partitionDef)
+		assert.NotNil(t, err)
+	}
+
+	// empty PartitionOptions
+	{
+		assert.NotNil(t, router)
+		_, err := router.ListUniform("t1", "i", sqlparser.PartitionOptions{})
+		assert.NotNil(t, err)
+	}
+}
+
+func TestRouterComputeList1(t *testing.T) {
```

Comment:
The testcase may execute fail, because the object `listMap` in function `ListUniform` is a map which is unordered. This may cause the table name to be unordered.  How about use slice type like `[]struct`.

Reply:
Good job.